bfvmm/include/hve/arch/intel_x64/vmexit/nmi_window.h

```diff
@@ -44,58 +41,87 @@
 #define EXPORT_HVE
 #endif
 
-namespace bfvmm::intel_x64::nmi
+// -----------------------------------------------------------------------------
+// Definitions
+// -----------------------------------------------------------------------------
+
+namespace bfvmm::intel_x64
 {
 
-/// NMI delegator
-///
-/// Delegates processing of non maskable interrupt based vmexits to registered
-/// handlers
+class vcpu;
+
+/// NMI window
 ///
-class EXPORT_HVE delegator
+class EXPORT_HVE nmi_window_handler
 {
-
 public:
 
-    /// Default Constructor
+    /// Constructor
+    ///
+    /// @expects
+    /// @ensures
     ///
-    /// @expects none
-    /// @ensures none
+    /// @param vcpu the vcpu object for this interrupt window handler
     ///
-    delegator() = default;
+    nmi_window_handler(gsl::not_null<vcpu *> vcpu);
 
     /// Destructor
     ///
-    /// @expects none
-    /// @ensures none
+    /// @expects
+    /// @ensures
     ///
-    ~delegator() = default;
+    ~nmi_window_handler() = default;
 
-    /// Handle
-    ///
-    /// Handle an NMI based vmexit using registered handlers
-    ///
-    /// @expects none
-    /// @ensures none
+public:
+
+    /// Queue NMI
     ///
-    /// @param vcpu The vcpu the vmexit occurred on
+    /// Queues an NMI for injection.
     ///
-    /// @return True if the vmexit was successfully handled, false otherwise
+    /// @expects
+    /// @ensures
     ///
-    bool handle_nmi(vcpu_t vcpu);
+    void queue_nmi();
 
-    /// Handle
-    ///
-    /// Handle an NMI window vmexit using registered handlers
+    /// Inject NMI
     ///
-    /// @expects none
-    /// @ensures none
+    /// Inject an NMI on the next VM entry.
     ///
-    /// @param vcpu The vcpu the vmexit occurred on
+    /// @expects
+    /// @ensures
     ///
-    /// @return True if the vmexit was successfully handled, false otherwise
+    /// @param vector the vector to inject into the guest
```

Comment:
delete

Reply:
Nice catch. This was removed in the second PR as doxygen found this too. 
Fixed